### DISCUSSION TITLE
How to restrict users from using modules in kata

### DISCUSSION THREAD

**Discussion by sapka1** [codewars/codewars.com#3311] [Q&A]
I try to create a kata and want to make it with a restriction of using modules. How can i make that? Help please

**Comment by hobovsky** [comment]
General idea is: do not block any modules. Blocking modules is difficult to get right, and tests which try to block modules usually end up unnecessarily convoluted and not even 90% cheat-proof. Additionally, challenges based on blocking some modules need a _really_ good reason to do so, otherwise they turn out to be just an artificial obstacle in an otherwise easy problem. A lot of things depends on language, and what exactly you want to block.

If you still want to try, you can ask for some hints on Codewars Discord and its `#help-author` channel.

**Comment by sapka1** [comment]
I want to make a permutation task, and there exists things like
itertools.permutation. But i want users to think, to use recursion or just
loop with condition.

пн, 2 груд. 2024 р., 22:45 користувач hobovsky ***@***.***>
пише:

**Comment by cececodes1** [comment]
You can use clear instructions in your kata description that says that the use of built in modules for permutations is not allowed. Emphasize that the goal is to implement the solution using recursion or loops. 

or design your test cases to validate the correctness of the permutations, and you can include edge cases that would be hard to handle without a proper understanding of recursion or loops. hope this helps :)
